classes/controller/FrontController.php

```diff
@@ -735,6 +735,13 @@ protected function displayMaintenancePage()
     {
         if ($this->maintenance == true || !(int) Configuration::get('PS_SHOP_ENABLE')) {
             $this->maintenance = true;
+
+            $is_admin = (int) (new Cookie('psAdmin'))->id_employee;
```

Comment:
Is this test enough to authenticate an admin in front ?

Reply:
`id_employee` is only set when the cookie is valid, it's set in https://github.com/PrestaShop/PrestaShop/blob/develop/controllers/admin/AdminLoginController.php#L256

It's also used here https://github.com/PrestaShop/PrestaShop/blob/develop/controllers/front/UploadController.php#L48